app/controllers/results_submission_controller.rb

```diff
@@ -44,6 +44,11 @@ def newcomer_dob_check
   end
 
   def compute_potential_duplicates
+    last_job_run = DuplicateCheckerJobRun.find_by(competition_id: params.require(:competition_id))
+    should_fail_last_job_run = last_job_run&.run_status_not_started? || last_job_run&.run_status_in_progress?
```

Comment:
This is a bit confusing to me, I had to read it twice to understand what you're doing.

My main source of confusion is this: If there is a `last_job_run`, and if it is marked as (for example) `in_progress`, then why should it fail?

The confusion is made even stronger based on the fact that this code snippet is executed no matter what. Maybe provide the user with some additional "override previous job run(s)" checkbox, that transmits a boolean to the backend? Or would that be too complicated to explain to the average Delegate?

Reply:
The checkbox might make it complicated to Delegate I assume. But irrespective of that, when a new run is executed, the results of the previous run is no longer relevant and cannot be fetched by the user as well. It's fine even if we don't 'fail' them because they won't be accessed any longer.

I'm marking it as fail just for the sake of completion. We don't know if the job will get completed in future or not, if it gets completed, it will get changed to success - but doesn't matter for us anyway.

Reply:
Then may I suggest a compromise, which is to introduce a new job status that describes "not sure whether it failed but we think it's running too long so we treat it as failed"?

That way, we can distinguish between jobs that actually failed, and jobs which we deprecated.